lib/content-api-client.js

```diff
@@ -0,0 +1,86 @@
+require('colors');
+const NetworkUtils = require('./utils/NetworkUtils');
+const {
+    renderedRegionsByPageTypeQuery,
+    renderedRegionsByPageTypeAndEntityIdQuery,
+} = require('./graphql/query');
+
+const networkUtils = new NetworkUtils();
+
+/**
+ * @param {object} options
+ * @param {string} options.accessToken
+ * @param {string} options.storeUrl
+ * @param {string} pageType
+ * @returns {Promise<renderedRegions: array}>}
+ */
+async function getRenderedRegionsByPageType({ accessToken, storeUrl, pageType }) {
+    try {
+        const query = renderedRegionsByPageTypeQuery(pageType);
+
+        const response = await networkUtils.sendApiRequest({
+            url: `${storeUrl}/graphql`,
+            headers: {
+                'cache-control': 'no-cache',
+                'content-type': 'application/json',
+                Authorization: `Bearer ${accessToken}`,
+            },
+            method: 'POST',
+            data: JSON.stringify({
+                query,
+            }),
+        });
+
+        return {
+            renderedRegions: response.data.data.site.content.renderedRegionsByPageType.regions,
```

Comment:
😨  can we possibly deconstruct this response?

Reply:
I am not sure if deconstruction would help as there is nothing else we need to extract except for regions 🤷‍♂️